sklearn/preprocessing/_discretization.py

```diff
@@ -166,6 +189,33 @@ def fit(self, X, y=None):
                 f" instead."
             )
 
+        # Check if subsample is needed
+        n_samples, n_features = X.shape
+
+        if self.strategy == "quantile" and self.subsample == "warn":
+            warnings.warn(
+                "In the future (vTBD) onwards subsample = 1e5"
+                "will be used by default. Pass subsample=None to"
+                "silence this warning for now.",
+                FutureWarning,
+            )
+
+        if self.subsample is not None and self.subsample != "warn":
```

Comment:
Currently this implementation always subsamples and not only for `strategy='quantile'`. We need to validate that if `subsample` is an integer, then `strategy=='quantile'`

Reply:
You need to take this comment into account as well.